tests/end2end/test_nic.py

```diff
@@ -241,3 +247,145 @@ def test_nic_crud(dpm_mode_cpcs):  # noqa: F811
                 part.delete()
             if adapter:
                 adapter.delete()
+
+
+def test_nic_backing_port_port_based(dpm_mode_cpcs):  # noqa: F811
+    # pylint: disable=redefined-outer-name
+    """
+    Test Nic.backing_port() for port-based NICs (e.g. RoCE, CNA).
+    """
+    if not dpm_mode_cpcs:
+        pytest.skip("HMC definition does not include any CPCs in DPM mode")
+
+    logger = setup_logging(LOGGING, 'test_nic_backing_port_port_based',
+                           LOG_FILE)
+    logger.debug("Entered test function")
+
+    cpc = random.choice(dpm_mode_cpcs)
```

Comment:
Hrmrm. On the one hand, I appreciate the idea to limit the execution to a single CPC, and to then pick that one randomly. 
But: random adds an element of random. So, worst case, your test passes / fails randomly (because of issues with the underlying CPCs). On the other hand, the underlying functionality should be stable, so "which CPC was picked" shouldn't have such effects.

Reply:
It is always possible that on machines used for FW development/test, something does not work. On stable machines (like our A224/T224/M224M in POK) it should work reliably. In the testcase, it is not practical to have knowledge about specific CPCs. In case something does not work, a user running end2end tests would need to use a different HMC. So I would say unstable development CPCs / HMCs is something we can deal with.